src/stream/stream/mod.rs

```diff
@@ -456,6 +458,54 @@ extension_trait! {
             Inspect::new(self, f)
         }
 
+        #[doc = r#"
+            Returns the last element of the stream.
+
+            # Examples
+
+            Basic usage:
+
+            ```
+            # fn main() { async_std::task::block_on(async {
+            #
+            use std::collections::VecDeque;
+
+            use async_std::prelude::*;
+
+            let s: VecDeque<usize> = vec![1, 2, 3].into_iter().collect();
+
+            let last  = s.last().await;
+            assert_eq!(last, Some(3));
+            #
+            # }) }
+            ```
+
+            An empty stream will return `None:
+            ```
+            # fn main() { async_std::task::block_on(async {
+            #
+            use std::collections::VecDeque;
+
+            use async_std::prelude::*;
+
+            let s: VecDeque<usize> = vec![].into_iter().collect();
+
+            let last  = s.last().await;
+            assert_eq!(last, None);
+            #
+            # }) }
+            ```
+            
+        "#]
+        fn last(
+            self,
+        ) -> impl Future<Output = Option<Self::Item>> + 'static [LastFuture<Self, Self::Item>]
```

Comment:
The `+ 'static` is not necessary (doesn't have an effect in this context) and we don't use it other similar methods :)

```suggestion
        ) -> impl Future<Output = Option<Self::Item>> [LastFuture<Self, Self::Item>]
```